### PR TITLE
info respects last rev flag

### PR DESCRIPTION
**Pull request by ittaiz**
Currently Info command ignores `--last-rev` while Migrate respects it.
I changed Info to work like Migrate.

### CODE REVIEW

File: java/com/google/copybara/Workflow.java [hunk 1]
```diff
@@ -228,6 +225,18 @@ public Info getInfo() throws RepoException, ValidationException {
       return Info.create(ImmutableList.of(migrationRef));
     });
   }
+  @Nullable
+  private String maybeGetLastRev() throws RepoException, ValidationException {
+    return (getLastRevisionFlag() != null) ?
+            getLastRevisionFlag() :
+            getPreviousRefFromOrigin();
```
Comment (mikelalcon) [inline java/com/google/copybara/Workflow.java]: Nit format and parenthesis :

return getLastRevisionFlag() != null 
    ? getLastRevisionFlag() 
    :  getPreviousRefFromOrigin();

File: javatests/com/google/copybara/WorkflowTest.java [hunk 2]
```diff
@@ -1211,6 +1213,55 @@ public void checkLastRevStatus_change_request() throws Exception {
...
+    // can add custom configuration to $HOME/.gitconfig.
+    options.setEnvironment(GitTestUtil.getGitEnv());
+    options.setHomeDir(Files.createTempDirectory("home").toString());
+    options.gitDestination.committerName = "Foo";
+    options.gitDestination.committerEmail = "foo@foo.com";
+    options.workflowOptions.checkLastRevState = true;
+    options.setLastRevision(firstCommit);
+
+    final Info<Revision> info = (Info<Revision>) loadConfig(config).getMigration("default").getInfo();
+    final List<String> commitMessages =
```
Comment (mikelalcon) [inline javatests/com/google/copybara/WorkflowTest.java]: You can replace this with:

assertThat(Iterables.getOnlyElement(info.migrationReferences()).getMessage()).isEquals("change1\n")

Comment (ittaiz) [inline javatests/com/google/copybara/WorkflowTest.java]: There's actually two levels of nesting there... `migrationReferences` and `getAvailableToMigrate`. I just code in Scala in my day to day so streams were easier. Is there a similar nifty thing for two levels of iterables?

File: java/com/google/copybara/Workflow.java (hunk 1 above)
Comment (ittaiz) [inline java/com/google/copybara/Workflow.java]: done

File: javatests/com/google/copybara/WorkflowTest.java (hunk 2 above)
Comment (mikelalcon) [inline javatests/com/google/copybara/WorkflowTest.java]: Ahh I didn't realize that was a nested list. Never mind :)

**Comment by ittaiz** [comment]
👍🏽 Anything else?
On Wed, 10 May 2017 at 22:11 Mikel Alcon <notifications@github.com> wrote:

> *@mikelalcon* commented on this pull request.
>
In javatests/com/google/copybara/WorkflowTest.java

**Comment by mikelalcon** [comment]
Squashed and merged in our internal git repo.

**Comment by ittaiz** [comment]
Thanks!

**Pull request closed without merging**